source/server/http_dynamic_delay_filter.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <atomic>
+#include <string>
+
+#include "envoy/server/filter_config.h"
+
+#include "api/server/response_options.pb.h"
+
+namespace Nighthawk {
+namespace Server {
+
+/**
+ * Filter configuration container class for the dynamic delay extension.
+ */
+class HttpDynamicDelayDecoderFilterConfig {
+
+public:
+  HttpDynamicDelayDecoderFilterConfig(nighthawk::server::ResponseOptions proto_config);
+
+  /**
+   * @return const nighthawk::server::ResponseOptions& read-only reference to the proto config
+   * object.
+   */
+  const nighthawk::server::ResponseOptions& server_config() { return server_config_; }
+
+  /**
+   * Increments the number of active instances.
+   */
+  void incrementInstanceCount() { instances()++; }
+  /**
+   * Decrements the number of active instances.
+   */
+  void decrementInstanceCount() { instances()--; }
+  /**
+   * @return uint64_t the approximate number of active instances.
+   */
+  uint64_t approximateInstances() const { return instances(); }
+
+private:
+  const nighthawk::server::ResponseOptions server_config_;
+  static std::atomic<uint64_t>& instances() {
+    // We lazy-init the atomic to avoid static initialization / a fiasco.
+    MUTABLE_CONSTRUCT_ON_FIRST_USE(std::atomic<uint64_t>, 0); // NOLINT
```

Comment:
As an alternative to this, I think you could examine the stats gauge maintained by Envoy tracking the number of currently active requests.

Reply:
For now this is implemented like with an atomic, for context see  https://github.com/envoyproxy/nighthawk/pull/390/files/3ec87c082f960576a489a696dbb2911877271653#diff-3b51180e80dfb0f4e9f97a9596c93bc1 and https://github.com/envoyproxy/envoy/issues/11799.